src/astra/astra.service.ts

```diff
@@ -0,0 +1,147 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { from, Observable } from 'rxjs';
+import {
+  deleteItem,
+  documentId,
+  findResult,
+} from '@cahllagerfeld/nestjs-astra';
+const DATASTAX_CLIENT = 'DATASTAX_CLIENT';
+
+@Injectable()
+export class AstraService {
+  private collection: any;
+  constructor(@Inject(DATASTAX_CLIENT) private readonly client: any) {}
+
+  private setupClient(namespace: string, collection: string) {
+    this.collection = this.client.namespace(namespace).collection(collection);
```

Comment:
I think we could try to save these into the collection with a unique key, but not sure if that helps with efficiency because it is a rest call.

Something like ...

```ts
private getClient(namespace: string, collection: string) {
   const key = `${namespace}-${collection}`;
   if (!this.collection[key]) {
      this.collection[key] = this.client.namespace(namespace).collection(collection);
   }

   return this.collection[key];
}
```